halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/AccountService.java

```diff
@@ -37,36 +38,42 @@
 @Component
 public class AccountService {
   @Autowired
-  ProviderService providerService;
+  LookupService lookupService;
 
   @Autowired
   ValidateService validateService;
 
-  public Account getAccount(NodeReference reference) {
-    Provider provider = providerService.getProvider(reference);
+  public List<Account> getAccounts(NodeReference reference) {
+    NodeFilter filter = new NodeFilter(reference).anyHalconfigFile().anyAccount();
 
-    String accountName = reference.getAccount();
+    return lookupService.getMatchingNodesOfType(filter, Account.class)
+            .stream()
+            .map(n -> (Account) n)
+            .collect(Collectors.toList());
+  }
 
-    List<Account> accounts = provider.getAccounts();
+  public Account getAccount(NodeReference reference) {
+    String accountName = reference.getAccount();
+    NodeFilter filter = new NodeFilter(reference).anyHalconfigFile();
 
-    List<Account> matchingAccounts = accounts
+    List<Account> matchingAccounts = lookupService.getMatchingNodesOfType(filter, Account.class)
         .stream()
-        .filter(a -> a.getName().equals(accountName))
+        .map(n -> (Account) n)
         .collect(Collectors.toList());
 
     switch (matchingAccounts.size()) {
       case 0:
         throw new IllegalRequestException(new ProblemBuilder(
-            Problem.Severity.FATAL, "No matching account found account")
+            Problem.Severity.FATAL, "No matching account with name \"" + accountName + "\" found")
```

Comment:
Right now, the only use-cases this endpoint enables are listing the accounts & validating them. Even when I add the ability to add/remove/edit accounts, specifiying a non-existing account for any of those operations is impossible to fulfill.